azurerm/resource_arm_media_services_account.go

```diff
@@ -0,0 +1,217 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/mediaservices/mgmt/2018-07-01/media"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMediaServicesAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMediaServicesAccountCreateUpdate,
+		Read:   resourceArmMediaServicesAccountRead,
+		Update: resourceArmMediaServicesAccountCreateUpdate,
+		Delete: resourceArmMediaServicesAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-a-z0-9]{3,24}$"),
+					"Media Services Account name must be 3 - 24 characters long, contain only lowercase letters and numbers.",
+				),
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"storage_account": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: azure.ValidateResourceID,
+						},
+
+						"is_primary": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+
+			"media_service_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
in retrospect looks like this field isn't being deserialized correctly; will push a commit to remove this

Reply:
upstream issue: https://github.com/Azure/azure-sdk-for-go/issues/4124